### ISSUE TITLE
WCAG 2.1 2.1.4 Character Key Shortcuts - does it include SPACE/ENTER ?

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag#666]
The normative text only mentions

> using only letter (including upper- and lower-case letters), punctuation, number, or symbol characters

Does this mean key shortcuts like SPACE and ENTER are explicitly excluded?

**Comment by patrickhlauke** [comment]
Also, more exotic keys like ESC/Escape, Home, End, Delete, F1, etc/ (which don't generate printable characters as such, but can presumably be triggered accidentally by voice recognition and can end up triggering JS functionality hanging off event listeners)

**Comment by alastc** [comment]
You just wanted issue 666, didn't you? ;-)

Non-official proposed response:

<hr>

Correct: By the wording and the intent it would not include space/enter/esc/home/delete/f1 etc.

Some of those would interfer with implementing general keyboard accessibility, and that was not intended. 

From the intent, the only one of those that might have an impact is space, I'm not sure if there is a difference between saying "press space" (intended) and "Hey Kim" with a space in the middle. 
It probably depends on what is in focus. But regardless, the intent was not to restrict the usage of non-printable character-keys for short-cuts.

**Comment by patrickhlauke** [comment]
fwiw, SPACE and ENTER do count as "printable characters" in JS terms. wondering if just saying "space" or "enter" on their own in speech recognition would also lead to those characters/keys being sent to the page and potentially triggering functionality.

**Comment by alastc** [comment]
It's worth checking with a voice-input expert (hello @KimPatch!), but if you say "space", that would be "s", "p" etc.

It isn't JavaScript per-se, so those characters don't print...

**Comment by patrickhlauke** [comment]
to give some context here, a site that has implemented a page-wide SPACE shortcut to play/pause some audio or video, but also has form controls like checkboxes and general buttons that a keyboard user would activate using SPACE - if the site doesn't "swallow" those key events, keyboard users end up triggering play/pause inadvertently. sure, this is more a failure of 2.1.1 in the broadest sense, but wondering if 2.1.4 also applies. And more generally, why those printable characters are excluded.

> Some of those would interfer with implementing general keyboard accessibility, and that was not intended.

I believe if those were implemented to actually trigger the correct functionality (e.g.  a faked button with a key event listener for SPACE), that's not so much a "shortcut" as an actual implementation of functionality...if that distinction makes any sense.

**Comment by KimPatch** [comment]
Yes, keys like Enter and Space are excluded on purpose. They are of no concern here. You can trip many letter keys at once by saying a single word or phrase. The letter keys are the issue for speech users. 

Cheers,
Kim

Kim Patch
www.redstartsystems.com

**Comment by patrickhlauke** [comment]
so saying "SPACE", "ENTER", etc fires the relevant individual characters, rather than triggering that particular key?

**Comment by patrickhlauke** [comment]
(additionally, i'd suggest adding a note or similar to the understanding document, explicitly stating that these other keys are exempt - either because they're non-printable characters, or because they're generally not triggered accidentally by voice command)

**Comment by detlevhfischer** [comment]
I had not included space in may draft Failure "Unmodifiable single key shortcut" https://github.com/w3c/wcag/pull/584 , which sports the ugliest test ever conceived. My rewording now says 

> Press all printing characters, i.e. all number, letter, sign and punctuation keys. Do not press non-printing modifier and control keys such as STR, ALT, Arrow keys as well as space, ENTER, Return, Tab, and Delete keys, and (where present) Function keys F1-F12.
> Hold the SHIFT key and press the same keys again.

So I have also included in exceptinons the Function keys and Delete.

**Comment by mbgower** [comment]
> so saying "SPACE", "ENTER", etc fires the relevant individual characters, rather than triggering that particular key?

@patrickhlauke  In Dragon you would have to issue a "Press" command to trigger Enter or Space keys (e.g., "Press Enter"). otherwise, if you just say "Enter", Dragon would just send the 5 character string E n t e r 
That is the whole point of avoiding single-character shortcuts. They can be unintentionally triggered in a stream so that the E may trigger an Edit command, for instance. Not so with any non-printable key.

**Comment by patrickhlauke** [comment]
FWIW in discussions around this SC, i'd avoid the use of the phrase "non-printable key", because SPACE and ENTER do count as printable (and are often referred to as such when discussing things like `keypress` events etc - see for instance https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode#Non-printable_keys_(function_keys) )

Wondering if the understanding doc could do with some explicit mention of Space/Enter...but i'll look at this separately. Going to close for now :)
